templates/common/infra/bicep/core/gateway/apim.bicep

```diff
@@ -0,0 +1,60 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+@description('The email address of the owner of the service')
+@minLength(1)
+param publisherEmail string = 'noreply@microsoft.com'
+
+@description('The name of the owner of the service')
+@minLength(1)
+param publisherName string = 'n/a'
+
+@description('The pricing tier of this API Management service')
+@allowed([
+  'Consumption'
+  'Developer'
+  'Standard'
+  'Premium'
+])
+param sku string = 'Consumption'
+
+@description('The instance size of this API Management service.')
+@allowed([ 0, 1, 2 ])
+param skuCount int = 0
+
+@description('Azure Application Insights Resource Id')
+param appInsightsResourceId string
+
+@description('Azure Application Insights Instrumentation Key')
+param appInsightsInstrumentationKey string
```

Comment:
since we have the instrumentationkey on the main.bicep file I thought that it will be easier just to pass the key as param

Reply:
I believe we are treating this key as a secret, which is why we don't pass as a param.  We could probably lighten that requirement since it's just a guid.

Reply:
I'm fixing it

Reply:
fixed